### DISCUSSION TITLE
How to watch files for different main file than package specifies?

### DISCUSSION THREAD

**Discussion by adjenks** [parcel-bundler/parcel#6147] [Q&A]
My use case is that I have two separate build configurations, a main one and an example project that uses the main file.

I want to `parcel watch` a file in an "example" directory which is an html file, but the package.json has an entry of "main":"main".ts". When I run my npm watch script, it says 
> Target "main" declares an output file path of "main.ts" which does not match the compiled bundle type "html".

So I want it to ignore my main entry and just build the thing that I gave it. How do I get it to ignore the main entry? I used `--dist-dir='/example/dist'` as mentioned [here](https://github.com/parcel-bundler/parcel/issues/3493) but it's still giving me that same error.

**Comment by adjenks** [comment]
It appears that if I use the bare `parcel` command it will build and serve it, but if I use `parcel build` or `parcel watch` it fails and says "did you mean?"

**Comment by mischnic** [comment]
`main / module / browser` are the **output** path for your bundle (e.g. when building libraries).

In most cases, you don't want a `"main"` at all.

It also has a higher precedence than `--dist-dir`, which is what that flag gets ignored.

Can you either remove `"main": "...."` or add `"targets": {"main": false}` to your package.json?

**Reply by adjenks** [reply]
I am making a library though. I just want to have a folder that contains a small web-application that demonstrates the use of the library. Parcel should be able to bundle both the library and the html file in the example folder.

**Reply by mischnic** [reply]
You could either

Add `"targets": {"default": {}}` to package.json and call `parcel example/index.html --target default`

or add a (empty) package.json to the example folder and invoke `parcel` inside of that folder

**Reply by adjenks** [reply]
Okay thank you. I suppose setting targets.main to false would work as well since I can still build it without that entry, but I created this because it makes sense for main to still be a target:
```json
  "targets": {
    "exampleapp": {
      "engines": {
        "browsers": "> 0.25%",
        "isLibrary": "false"
      }
    }
  },
```
and call this `--target=exampleapp` and it seems to work.
However, if I set isLibrary to `false` without quotes, it says it's expecting a string, but the docs say it should be a boolean.

Also, can I put "targets" into .parcelrc? I'd rather keep non-npm stuff out of package.json if possible.
Hmm. I just tried. It appears not. Too bad.

**Comment by ELI7VH** [comment]
I would also prefer that parcel settings could be confined to a `.parcelrc` it just makes sense

**Reply by adjenks** [reply]
Further discussion going on here: https://github.com/parcel-bundler/parcel/issues/5243
